pyccl/tests/test_profiles.py

```diff
@@ -0,0 +1,231 @@
+
+    p = prof_class(c)
+    smoke_assert_prof_real(p)
+
+
+@pytest.mark.parametrize('prof_class',
+                         [ccl.halos.HaloProfileGaussian,
+                          ccl.halos.HaloProfilePowerLaw])
+def test_simple_smoke(prof_class):
+    def r_s(cosmo, M, a, mdef):
+        return mdef.get_radius(cosmo, M, a)
+
+    p = prof_class(r_s, one_f)
+    smoke_assert_prof_real(p)
+
+
+def test_gaussian_accuracy():
+    def fk(k):
+        return np.pi**1.5 * np.exp(-k**2 / 4)
+
+    p = ccl.halos.HaloProfileGaussian(one_f, one_f)
+
+    k_arr = np.logspace(-3, 2, 1024)
+    fk_arr = p.fourier(COSMO, k_arr, 1., 1.)
+    fk_arr_pred = fk(k_arr)
+    res = np.fabs(fk_arr - fk_arr_pred)
+    assert np.all(res < 5E-3)
+
+
+@pytest.mark.parametrize('alpha', [-1.2, -2., -2.8])
+def test_projected_plaw(alpha):
+    from scipy.special import gamma
+
+    prefac = (np.pi**0.5 * gamma(-(alpha + 1) / 2) /
+              gamma(-alpha / 2))
+
+    def s_r_t(rt):
+        return prefac * rt**(1 + alpha)
+
+    def alpha_f(cosmo, a):
+        return alpha
+
+    p = ccl.halos.HaloProfilePowerLaw(one_f, alpha_f)
+    p.update_precision_fftlog(plaw_index=alpha)
+
+    rt_arr = np.logspace(-3, 2, 1024)
+    srt_arr = p.projected(COSMO, rt_arr, 1., 1.)
+    srt_arr_pred = s_r_t(rt_arr)
+    res = np.fabs(srt_arr / srt_arr_pred - 1)
+    assert np.all(res < 5E-3)
+
+
+@pytest.mark.parametrize('alpha', [-1.2, -2., -2.8])
+def test_plaw_accuracy(alpha):
+    from scipy.special import gamma
+
+    prefac = (2.**(3+alpha) * np.pi**1.5 *
+              gamma((3 + alpha) / 2) /
+              gamma(-alpha / 2))
+
+    def fk(k):
+        return prefac / k**(3 + alpha)
+
+    def alpha_f(cosmo, a):
+        return alpha
+
+    p = ccl.halos.HaloProfilePowerLaw(one_f, alpha_f)
+    p.update_precision_fftlog(plaw_index=alpha)
+
+    k_arr = np.logspace(-3, 2, 1024)
+    fk_arr = p.fourier(COSMO, k_arr, 1., 1.)
+    fk_arr_pred = fk(k_arr)
+    res = np.fabs(fk_arr / fk_arr_pred - 1)
+    assert np.all(res < 5E-3)
+
+
+@pytest.mark.parametrize("use_analytic", [True, False])
+def test_nfw_accuracy(use_analytic):
+    from scipy.special import sici
+
+    tol = 1E-10 if use_analytic else 5E-3
+    cM = ccl.halos.ConcentrationDuffy08(M200)
+    p = ccl.halos.HaloProfileNFW(cM, fourier_analytic=use_analytic)
+    M = 1E14
+    a = 0.5
+    c = cM.get_concentration(COSMO, M, a)
+    r_Delta = M200.get_radius(COSMO, M, a) / a
+    r_s = r_Delta / c
+
+    def fk(k):
+        x = k * r_s
+        Si1, Ci1 = sici((1 + c) * x)
+        Si2, Ci2 = sici(x)
+        P1 = 1 / (np.log(1+c) - c / (1 + c))
+        P2 = np.sin(x) * (Si1 - Si2) + np.cos(x) * (Ci1 - Ci2)
+        P3 = np.sin(c * x)/((1 + c) * x)
+        return M * P1 * (P2 - P3)
+
+    k_arr = np.logspace(-2, 2, 256) / r_Delta
+    fk_arr = p.fourier(COSMO, k_arr, M, a, M200)
+    fk_arr_pred = fk(k_arr)
+    res = np.fabs((fk_arr - fk_arr_pred) / M)
+    assert np.all(res < tol)
```

Comment:
A comment on what you are doing here would be helpful. Why divide by `M`?

Reply:
Done. This is basically normalizing the profile to 1 at large scales. Computing the relative error becomes too stringent at high k, where the profile becomes very small and you're almost dividing by zero.